tools/functional-tester/etcd-tester/checks.go

```diff
@@ -24,14 +24,18 @@ type Checker interface {
 	Check() error
 }
 
+type hashAndRevGetter interface {
+	getRevisionHash() (map[string]int64, map[string]int64, error)
+}
+
 type hashChecker struct {
-	tt *tester
+	hrg hashAndRevGetter
 }
 
-func newHashChecker(tt *tester) Checker { return &hashChecker{tt} }
+func newHashChecker(c *cluster) Checker { return &hashChecker{hashAndRevGetter(c)} }
```

Comment:
change this to newHashChecker(hrg hashAndRevGetter) convert cluster to hrg outside the new func. it is cleaner.